locale/en/docs/guides/security/index.md

```diff
@@ -0,0 +1,427 @@
+network access.
+Furthermore, because they also have access to the file system, they can send
+any data to anywhere.
+
+All code running into a node process has the ability to load and run additional
+arbitrary code using /p;=(or its equivalents).
+All code with file system write access may achieve the same thing by writing to
+new or existing files which are loaded.
+
+**This kind of attack is not considered a vulnerability in Node.js, rather it
+is the responsibility of the application to ensure its dependencies are safe
+and correct.**
+
+Node.js has a [policy mechanism][] to declare the loaded resource as
+untrusted or trusted.
+However, this policy is not enabled by default.
+Ensure to pin dependency versions and run automatic checks for vulnerabilities
+using common workflows or npm scripts.
+Before installing a package make sure that this package is maintained and
+includes all the content you expected in.
+Be careful, the Github source code is not always the same as the published one,
+validate it in the _node_modules_.
+
+#### Supply chain attacks
+
+A supply chain attack on a Node.js application happens when one of its
+dependencies (either direct or transitive) is compromised.
+This can happen either due to the application being too lax on the specification
+of the dependencies (allowing for unwanted updates) and/or common typos in the
+specification (vulnerable to [typosquatting][]).
+
+An attacker that takes control of an upstream package can publish a new version
+with malicious code in it. If a Node.js application depends on that package
+without being strict on which version is safe to use, the package can be
+automatically updated to the latest malicious version, compromising the application.
+
+Dependencies specified in the `package.json` file can have an exact version number
+or a range. However, when pinning a dependency to an exact version, its
+transitive dependencies are not themselves pinned.
+This still leaves the application vulnerable to unwanted/unexpected updates.
+
+Possible vector attacks:
+
+* Typosquatting attacks
+* Lockfile poisoning
+* Compromised maintainers
+* Malicious Packages
+* Dependency Confusions
+
+**Mitigation**
+
+* Using native modules with prebuilds to avoid native buildings, and verifying SHAs
+* Prevent npm from executing arbitrary scripts / Disabling npm postinstall
+* Pin dependency versions
+* Use lockfiles, which pin every dependency (direct and transitive).
+  * Use [Mitigations for lockfile poisoning][].
+* Automate checks for new vulnerabilities using CI, with tools like npm-audit.
+* Use [`npm ci`][] instead of `npm install`.
+This enforces the lockfile so that inconsistencies between it and the
+_package.json_ file cause an error (instead of silently ignoring the lockfile
+in favor of _package.json_).
+* Use [`npm audit`][] in CI to frequently scan for vulnerabilities in your
+dependencies.
+* Carefully check the _package.json_ file for errors/typos in the names of the
+dependencies.
```

Comment:
Can we add a bullet here for thoroughly auditing the code in third-party dependencies, whether with a human review (the gold standard) or at least using an automated tool in CI (e.g. Socket) which can detect unexpected package behavior, e.g. unneeded network or filesystem access?

Reply:
This could likely be expanded a bunch but might be out of scope for node core. Things like bundledDependencies, remote dependencies, and lock files can make package.json mislead audits. Idk depth this should go into.

Reply:
I recommend Socket (https://socket.dev) which analyzes npm packages with static analysis to find risky behaviors such as network or filesystem access, which aids when auditing third-party dependencies. (Disclosure: I'm working on this tool)

Examples of Socket catching malware:

- https://socket.dev/npm/package/webb3/overview/1.2.0
- https://socket.dev/npm/package/webpback-dev-esrver/overview/1.2.0